cmd/operator-sdk/build/cmd.go

```diff
@@ -93,8 +93,6 @@ func buildFunc(cmd *cobra.Command, args []string) error {
 
 	if value, ok := os.LookupEnv("GOARCH"); ok {
 		goBuildEnv = append(goBuildEnv, "GOARCH="+value)
-	} else {
-		goBuildEnv = append(goBuildEnv, "GOARCH=amd64")
 	}
```

Comment:
This entire section can be removed. `goBuildEnv` is setup from the current environment int line 92, so line 95 doesn't actually do anything. Either GOARCH is already set in the environment, or it is unset, in which case go will use the default GOARCH (based on the architecture of the `go` binary itself).

Reply:
Also probably need a CHANGELOG update for this change.

I'm thinking a bugfix:
```
Updates `operator-sdk build` for go operators to compile the operator binary based on Go's built-in GOARCH detection. This fixes an issue that caused an `amd64` binary to be built into non-`amd64` base images when using operator-sdk on non-`amd64` architectures. ([#2268](https://github.com/operator-framework/operator-sdk/pull/2268))
```

Reply:
Updated and tested @joelanford